### ISSUE TITLE
Google Sheet range must include Row 'A' to catch headers

### ISSUE THREAD

**Issue by McFateM** [mnylc/islandora_multi_importer#46]
In my tests I have found that the "range" specification required when reading from a Google Sheet MUST begin from cell A1 (as in Sheet1!A1:AN100).  If it does not, the column headers are not processed causing many exceptions downstream.

Unfortunately, I'm not sure how to fix this yet.

**Comment by DiegoPino** [comment]
Best fix is 1) Documentation! And catching the fact that if you don’t get a header row the rest won’t work. I would avoid thinking we can actually clean everything and being smarter than our users, or, always fetch row A1..

Diego Pino Navarro
Metropolitan New York Library Council
599 11th Av. New York, NY 10036

**Comment by McFateM** [comment]
Agreed.  I’m thinking it would not be hard to do all three… document, force Google Sheet reading of row ‘A’ regardless of range specification, and catch the exception if headers are not defined (although I’m not entirely sure what to look for in order to detect this reliably).

**Comment by DiegoPino** [comment]
Up for a call on Wednesday? I can do that for sure. Even reading twice, first ROW always + anything else people define. If ROW 1 is included, then well, only read once. Let me know if you have time on Wednesday

Best!

**Comment by McFateM** [comment]
@DiegoPino Did we ever discuss this, I can't recall?   Anyway, I've found an acceptable way around this in my workflow... by copying my row 'A' headers into the first row that I intend to read.  So, I agree, documentation is probably what's really required at this point. 

That makes me wonder, where should this be documented?  I think the ICG still has a fork of IMI with quite a bit of user documentation that was generated a year ago and our 2017 Hack/Doc.  If I get some time I'll take a look back at that and maybe we can discuss steps to create a PR from it?
